subsys/bluetooth/host/hci_core.c

```diff
@@ -3373,13 +3373,33 @@ void hci_event_prio(struct net_buf *buf)
 	}
 }
 
+#if !defined(CONFIG_BT_RECV_IS_RX_THREAD)
+static void rx_queue_put(struct net_buf *buf)
+{
+	int err;
+
+	net_buf_put(&bt_dev.rx_queue, buf);
+
+	err = k_work_submit(&rx_work);
+	if (err < 0) {
+		BT_ERR("Could not submit rx_work: %d", err);
+	}
+}
+#endif
+
 k_tid_t bt_recv_thread_id;
 
 int bt_recv(struct net_buf *buf)
 {
+#if defined(CONFIG_BT_RECV_IS_RX_THREAD)
 	if (bt_recv_thread_id == NULL) {
 		bt_recv_thread_id = k_current_get();
 	}
+#else
+	if (bt_recv_thread_id == NULL) {
+		bt_recv_thread_id = &k_sys_work_q.thread;
```

Comment:
I think it's a bugfix here, because it wasn't assigned a value before, which would cause bt_recv_thread_id to never be equal to BT RX.

Reply:
From `hci_core.c`:
```c
int bt_recv(struct net_buf *buf)
{
	if (bt_recv_thread_id == NULL) {
		bt_recv_thread_id = k_current_get();
	}
```

Reply:
I don't think so, according PR(https://github.com/zephyrproject-rtos/zephyr/pull/40056), It doesn't take into account that when this `CONFIG_BT_RECV_IS_RX_THREAD=n`.

Reply:
bt_recv may be call for `h4.c` or `h5.c` or `RPMsg` etc..

https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/att.c#L3240

Will always `false`  when `CONFIG_BT_RECV_IS_RX_THREAD=n`.

Reply:
I'm confused. I believe `CONFIG_BT_RECV_IS_RX_THREAD` does not matter here. `bt_recv_thread_id` is used to determine if `bt_recv()` is on call stack. Are you saying [ATT request allocation] can deadlock if `CONFIG_BT_RECV_IS_RX_THREAD=n`?:

[ATT request allocation]: https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/host/att.c#L3248

Reply:
yes

Reply:
The bt_recv_thread_id you added here does not represent BT RX in hci_core.c. If CONFIG_BT_RECV_IS_RX_THREAD=n, ATT Requst Allocation in BT RX will still cause deadlock.

Reply:
When CONFIG_BT_RECV_IS_RX_THREAD=n, all data will be processed by BT RX through net_buf_put, including ACL, so we should also consider this situation.

Reply:
I see. I'll open a new issue.

Reply:
Yes, I tried to fix it together with this pr, and in a separate commit, just to confirm with you.

Reply:
Here is the issue: #43448